### ISSUE TITLE
Support pyproject.toml as well as setup.cfg in kolibri-tools

### ISSUE THREAD

**Issue by pwithnall** [learningequality/kolibri#11146]
## Observed behavior

kolibri-tools currently [hard-codes a search](https://github.com/learningequality/kolibri/blob/2283fc4485e9533dcb95f9b52ce7fa19fc7ee806/packages/kolibri-tools/lib/cli.js#L27) for a Python-style `setup.cfg` in the project’s directory, to load locale configuration from.

As per [PEP-621](https://peps.python.org/pep-0621/), `pyproject.toml` is now a [recommended replacement](https://ianhopkinson.org.uk/2022/02/understanding-setup-py-setup-cfg-and-pyproject-toml-in-python/) for `setup.cfg`, though both are going to be supported for a while. A number of tools now support `pyproject.toml`, and others don’t.

## Expected behavior

It would be useful if kolibri-tools supported loading locale configuration from `pyproject.toml`, falling back to `setup.cfg` as before. Then it could be used in projects (Kolibri plugins) which use `pyproject.toml`.

The section name it uses would likely have to be something like `tool.kolibri.i18n` rather than `kolibri:i18n`, to fit in with `pyproject.toml` conventions.

kolibri-explore-plugin [uses `pyproject.toml`](https://github.com/endlessm/kolibri-explore-plugin/blob/master/pyproject.toml), for example.

## User-facing consequences

None, this is pretty low priority.

## Steps to reproduce

Run `kolibri-tools i18n-extract-messages …` on a Kolibri plugin project which uses `pyproject.toml`, and it will fail with
```
Kolibri: Getting files from glob: …/kolibri-explore-plugin/kolibri_explore_plugin/**/*.@(vue|js)
Kolibri: Processing  56  files...
Kolibri: Getting files from glob: packages/welcome-screen/**/*.@(vue|js)
Kolibri: Processing  19  files...
node:internal/validators:120
    throw new ERR_INVALID_ARG_TYPE(name, 'string', value);
    ^

TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
    at new NodeError (node:internal/errors:371:5)
    at validateString (node:internal/validators:120:11)
    at Object.join (node:path:1172:7)
    at module.exports (…/kolibri-explore-plugin/node_modules/kolibri-tools/lib/i18n/ExtractMessages.js:131:24)
    at Command.<anonymous> (…/kolibri-explore-plugin/node_modules/kolibri-tools/lib/cli.js:570:3)
    at Command.listener [as _actionHandler] (…/kolibri-explore-plugin/node_modules/kolibri-tools/node_modules/commander/lib/command.js:482:17)
    at …/kolibri-explore-plugin/node_modules/kolibri-tools/node_modules/commander/lib/command.js:1283:65
    at Command._chainOrCall (…/kolibri-explore-plugin/node_modules/kolibri-tools/node_modules/commander/lib/command.js:1177:12)
    at Command._parseCommand (…/kolibri-explore-plugin/node_modules/kolibri-tools/node_modules/commander/lib/command.js:1283:27)
    at …/kolibri-explore-plugin/node_modules/kolibri-tools/node_modules/commander/lib/command.js:1081:27 {
  code: 'ERR_INVALID_ARG_TYPE'
}
error Command failed with exit code 1.
```

**Comment by rtibbles** [comment]
Main thing would be finding a decent NodeJS toml parser, this is the most recently updated on npm: https://www.npmjs.com/package/@ltd/j-toml

**Comment by rtibbles** [comment]
Alternatively if Python can natively parse TOML, could call out to Python similarly to how we do for webpack JSON generation.

**Comment by pwithnall** [comment]
I have some WIP work to implement it here: https://github.com/endlessm/kolibri/tree/11146-pyproject.toml

but I won’t be able to pick it up and finish it off for a few days at least, so if someone else wants to pick it up in the meantime they are very welcome. I haven’t tested the toml parsing yet, for example.

**Comment by pwithnall** [comment]
OK, I’ve got a tested PR together here: https://github.com/learningequality/kolibri/pull/11227

**Comment by rtibbles** [comment]
Fixed in #11227
